crates/transaction-pool/src/pool/pending.rs

```diff
@@ -342,14 +334,23 @@ impl<T: TransactionOrdering> PendingPool<T> {
         let tx = self.by_id.remove(id)?;
         self.size_of -= tx.transaction.size();
 
-        if let Some(highest) = self.highest_nonces.get(&id.sender) {
-            if highest.transaction.nonce() == id.nonce {
+        if let Some(highest) = self.highest_nonces.get(&id.sender) &&
+            highest.transaction.nonce() == id.nonce
+        {
+            if let Some((_, new_highest)) = self
+                .by_id
+                .range((
+                    id.sender.start_bound(),
+                    std::ops::Bound::Included(TransactionId::new(id.sender, u64::MAX)),
+                ))
+                .last()
+            {
+                self.highest_nonces.insert(id.sender, new_highest.clone());
+            } else {
                 self.highest_nonces.remove(&id.sender);
             }
```

Comment:
I don't think this is correct, because this would remove the highest entry if the removed tx isnt the highest?